tests/Export/ExportItemAdapterTest.php

```diff
@@ -0,0 +1,179 @@
+use Shopware\Core\Framework\Test\TestCaseBase\IntegrationTestBehaviour;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Throwable;
+
+class ExportItemAdapterTest extends TestCase
+{
+    use IntegrationTestBehaviour;
+    use SalesChannelHelper;
+    use ProductHelper;
+
+    private $loggerMock = null;
+
+    protected function setUp(): void
+    {
+        $this->salesChannelContext = $this->buildSalesChannelContext();
+        $this->getContainer()->set('fin_search.sales_channel_context', $this->salesChannelContext);
+        $this->getContainer()->set(
+            'fin_search.export_context',
+            new ExportContext(
+                'ABCDABCDABCDABCDABCDABCDABCDABCD',
+                [],
+                $this->getCategory()
+            )
+        );
+        DynamicProductGroupService::getInstance(
+            $this->getContainer(),
+            $this->getContainer()->get('serializer.mapping.cache.symfony'),
+            Context::createDefaultContext(),
+            'ABCDABCDABCDABCDABCDABCDABCDABCD',
+            0
+        );
+
+        $this->loggerMock = $this->getMockBuilder(Logger::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        parent::setUp();
+    }
+
+    public function testProductInvalidExceptionIsLogged(): void
+    {
+        $id = Uuid::randomHex();
+
+        $productEntity = $this->createTestProduct([
+            'id' => $id,
+            'categories' => []
+        ]);
+
+        $expectedMessage = sprintf(
+            'Product "%s" with id %s was not exported because it has no categories assigned',
+            $productEntity->getTranslation('name'),
+            $productEntity->getId()
+        );
+
+        $this->expectAdapterException(
+            $productEntity,
+            new ProductHasNoCategoriesException($productEntity),
+            $expectedMessage
+        );
+    }
+
+    public function testEmptyValueIsNotAllowedExceptionIsLogged(): void
+    {
+        $id = Uuid::randomHex();
+
+        $productEntity = $this->createTestProduct([
+            'id' => $id
+        ]);
+
+        $error = sprintf(
+            'Product "%s" with id "%s" could not be exported.',
+            $productEntity->getTranslation('name'),
+            $productEntity->getId()
+        );
+        $reason = 'It appears to have empty values assigned to it.';
+        $help = 'If you see this message in your logs, please report this as a bug.';
+        $expectedMessage = implode(' ', [$error, $reason, $help]);
+
+        $this->expectAdapterException(
+            $productEntity,
+            new EmptyValueNotAllowedException(''),
+            $expectedMessage
+        );
+    }
+
+    public function testThrowableExceptionIsLogged(): void
+    {
+        $id = Uuid::randomHex();
+        $errorMessage = 'This product failed, because it is faulty.';
+
+        $productEntity = $this->createTestProduct([
+            'id' => $id
+        ]);
```

Comment:
The ID is never used, as you always get it with `$productEntity->getId()` anyway

```suggestion
        $errorMessage = 'This product failed, because it is faulty.';

        $productEntity = $this->createTestProduct();
```